client/jetpack-onboarding/steps/business-address.jsx

```diff
@@ -23,13 +23,13 @@ import { JETPACK_ONBOARDING_STEPS as STEPS } from '../constants';
 import { saveJetpackOnboardingSettings } from 'state/jetpack-onboarding/actions';
 
 class JetpackOnboardingBusinessAddressStep extends React.PureComponent {
-	state = {
-		city: '',
-		name: '',
-		state: '',
-		street: '',
-		zip: '',
-	};
+	state = get( this.props.settings, 'businessAddress' );
+
+	componentWillReceiveProps( nextProps ) {
+		if ( this.props.isRequestingSettings && ! nextProps.isRequestingSettings ) {
+			this.setState( nextProps.settings.businessAddress );
```

Comment:
We need to skip this in the default case where business address is empty and the `businessAddress` option returned by the API is `false`. In that case, we'll still need the default empty string values that are being removed here.